.gitlab/build-oci.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+source common_build_functions.sh
+
+if [ -n "$CI_COMMIT_TAG" ] && [ -z "$DOTNET_PACKAGE_VERSION" ]; then
+  DOTNET_PACKAGE_VERSION=${CI_COMMIT_TAG##v}
+fi
+
+if [ -z "$ARCH" ]; then
+  ARCH=amd64
+fi
+
+TMP_DIR=$(mktemp --dir)
+
+curl --location --fail \
+  --output $TMP_DIR/datadog-dotnet-apm.old \
+  "https://github.com/DataDog/dd-trace-dotnet/releases/download/v$DOTNET_PACKAGE_VERSION/datadog-dotnet-apm_${DOTNET_PACKAGE_VERSION}_$ARCH.deb"
+
+fpm --input-type deb \
+  --output-type dir \
+  --name datadog-dotnet-apm \
+  $TMP_DIR/datadog-dotnet-apm.old
+
+echo -n $DOTNET_PACKAGE_VERSION > $TMP_DIR/opt/datadog/version/auto_inject-dotnet.version
+
+# Build packages
+datadog-package create \
```

Comment:
I'd love to know exactly _why_ this is necessary, and whether we can update our existing packages to make the re-packing unneccessary... lib-injection is already doing repackaging and I wonder if we could settle on a standard format 🤔

Reply:
The main reason we need a new package format is that we are working on the auto update of APM libraries for host setups. This is going to be used in conjunction with lib-injection. The goal is for the OCI package to be the standard format in the future.

This auto update mechanism works by having the following fs structure
/opt/datadog-packages/<package name like dd-trace-dotnet>/<version>
                                                                                                          /stable      <- symlink to the current version
 and a process called the updater, connected to Remote Config that can download and unpack new versions.
To make this work nicely, we need a platform independent package format, that can be unpacked in a single relative path, which existing debs and rpm don't offer.